File: simcoe/src/main/java/com/prolificinteractive/simcoe/TrackingResult.java
package com.prolificinteractive.simcoe;

/**
 * The tracking result.
 */
public class TrackingResult {

  private final boolean error;
  private final String errorMessage;

  /**
   * Creates a tracking result instance.
   *
   * @param errorMessage The error message, or null.
   */
  public TrackingResult(final String errorMessage) {
    error = errorMessage != null;
    this.errorMessage = errorMessage;
  }

  /**
   * @return Error message.
   */
  public String getErrorMessage() {
    return errorMessage;
  }

  /**
   * Whether the tracking result contains an error message.
   */
  public boolean isError() {
    return error;
  }
}

File: simcoe/src/main/java/com/prolificinteractive/simcoe/TrackingResult.kt
package com.prolificinteractive.simcoe

/**
 * The tracking result.
 *
 * @param errorMessage Error message.
 */
open class TrackingResult(val errorMessage: String = "") {

  /**
   * Whether the tracking result contains an error message.
   */
  val isError: Boolean = errorMessage.isNotEmpty()

}

class Success : TrackingResult()

class Failure(errorMessage: String) : TrackingResult(errorMessage)